src/Simulation/Simulators/QuantumSimulator/StateDumper.cs

```diff
@@ -60,29 +67,190 @@ public virtual bool Dump(IQArray<Qubit>? qubits = null)
+            /// <param name="convention">
+            ///     The convention to be used in labeling each computational basis state.
+            /// </param>
+            /// <param name="truncateSmallAmplitudes">
+            ///     Whether to truncate small amplitudes.
+            /// </param>
+            /// <param name="truncationThreshold">
+            ///     If <paramref name="truncateSmallAmplitudes" /> is <c>true</c>,
+            ///     then amplitudes whose absolute value squared are below this
+            ///     threshold are suppressed.
+            /// </param>
+            public IEnumerable<(Complex, string)> SignificantAmplitudes(
+                BasisStateLabelingConvention convention,
+                bool truncateSmallAmplitudes, double truncationThreshold
+            ) =>
+                (
+                    truncateSmallAmplitudes
+                    ? Amplitudes
+                        .Select((amplitude, idx) => (amplitude, idx))
+                        .Where(item =>
+                            System.Math.Pow(item.amplitude.Magnitude, 2.0) >= truncationThreshold
+                        )
+                    : Amplitudes.Select((amplitude, idx) => (amplitude, idx))
+                )
+                .Select(
+                    item => (item.amplitude, BasisStateLabel(convention, item.idx))
+                )
+                .OrderBy(
+                    item => item.Item2,
+                    // If a basis state label is numeric, we want to compare
+                    // numerically rather than lexographically.
+                    convention switch {
+                        BasisStateLabelingConvention.BigEndian => ToIntComparer,
+                        BasisStateLabelingConvention.LittleEndian => ToIntComparer,
+                        _ => Comparer<string>.Default
+                    }
+                );
+
+            /// <summary>
+            ///     Using the given labeling convention, returns the label for a
+            ///     computational basis state described by its bit string as encoded
+            ///     into an integer index in the little-endian encoding.
+            /// </summary>
+            public string BasisStateLabel(
+                BasisStateLabelingConvention convention, int index
+            ) => convention switch
+                {
+                    BasisStateLabelingConvention.Bitstring =>
+                        String.Concat(
+                            System
+                                .Convert
+                                .ToString(index, 2)
+                                .PadLeft(NQubits, '0')
+                                .Reverse()
+                        ),
+                    BasisStateLabelingConvention.BigEndian =>
+                        System.Convert.ToInt64(
+                            String.Concat(
+                                System.Convert.ToString(index, 2).PadLeft(NQubits, '0').Reverse()
```

Comment:
The case above should probably also be written in one line.

Reply:
Makes sense. If I'll be making more changes (with subsequent CI rebuild) then I'll make this change.
But if the build succeeds then I'd prefer to merge.